File: depth_sensor/src/DepthSensorPublisher.cc
#include "depth_sensor/DepthSensorPublisher.hh"

namespace MuddSub::DepthSensor
{
  DepthSensorPublisher::DepthSensorPublisher(ros::NodeHandle n):
    n_(n)
  {
    depth_pub = n_.advertise<depth_sensor::Depth>("depth_sensor/depth", 1000);
    pressure_pub = n_.advertise<sensor_msgs::FluidPressure>("depth_sensor/pressure", 1000);
    temperature_pub = n_.advertise<sensor_msgs::Temperature>("depth_sensor/temperature", 1000);
  }

  void DepthSensorPublisher::publishDepth(depth_sensor::Depth& depth)
  {
    depth_pub.publish(depth);
  }

  void DepthSensorPublisher::publishPressure(sensor_msgs::FluidPressure pressure)
  {
    pressure_pub.publish(pressure);
  }

  void DepthSensorPublisher::publishTemperature(sensor_msgs::Temperature temperature)
  {
    temperature_pub.publish(temperature);
  }
}

File: depth_sensor/scripts/DepthSensorPublisher.py
#!/usr/bin/env python

import rospy
from depth_sensor.msg import Depth
from sensor_msgs.msg import FluidPressure
from sensor_msgs.msg import Temperature

class DepthSensorPublisher:
    def __init__(self):
        self.wrench_pub = rospy.Publisher('depth_sensor/depth', Depth, queue_size=10)
        self.pressure_pub = rospy.Publisher('depth_sensor/pressure', FluidPressure, queue_size=10)
        self.temperature_pub = rospy.Publisher('depth_sensor/temperature', Temperature, queue_size=10)

    def publishDepth(self, depth):
        self.wrench_pub.publish(depth)

    def publishPressure(self, pressure):
        self.pressure_pub.publish(pressure)

    def publishTemperature(self, temperature):
        self.temperature_pub.publish(temperature)
